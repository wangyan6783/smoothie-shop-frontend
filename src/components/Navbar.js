import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";

function Navbar(props) {
  function renderLogin() {
    if (props.currentUser) {
      console.log("signedin");
      return (
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Sign Out
          </Link>
        </li>
      );
    } else {
      return (
        <li className="navbar-item">
          <Link className="navbar-link" to="/login">
            Login
          </Link>
        </li>
      );
    }
  }

  return (
    <nav className="navbar">
      <ul className="navbar-list">
        <li className="navbar-item">
          <Link className="navbar-link" to="/smoothies">
            Browse
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Our Story
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Gifts
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-logo" to="/">
            Daily Harvest
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Help
          </Link>
        </li>
        {renderLogin()}
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Get Started
          </Link>
        </li>
      </ul>
    </nav>
  );
}

const mapStateToProps = state => ({
  currentUser: state.user.currentUser
});

export default connect(mapStateToProps)(Navbar);
