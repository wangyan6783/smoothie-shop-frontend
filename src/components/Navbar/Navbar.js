import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { setCurrentUser } from "../../redux/user/user.actions";

function Navbar(props) {
  function renderLogin() {
    if (props.currentUser) {
      return (
        <li className="navbar-item">
          <Link className="navbar-link" to="/" onClick={props.signout}>
            Sign Out
          </Link>
        </li>
      );
    } else {
      return (
        <li className="navbar-item">
          <Link className="navbar-link" to="/login">
            Login
          </Link>
        </li>
      );
    }
  }

  return (
    <nav className="navbar">
      <ul className="navbar-list">
        <li className="navbar-item">
          <Link className="navbar-link" to="/smoothies">
            Browse
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Our Story
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Gifts
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-logo" to="/">
            Daily Harvest
          </Link>
        </li>
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Help
          </Link>
        </li>
        {renderLogin()}
        <li className="navbar-item">
          <Link className="navbar-link" to="/">
            Get Started
          </Link>
        </li>
      </ul>
    </nav>
  );
}

const mapStateToProps = state => ({
  currentUser: state.user.currentUser
});

const mapDispatchToProps = dispatch => ({
  signout: () => dispatch(setCurrentUser(null))
});

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
