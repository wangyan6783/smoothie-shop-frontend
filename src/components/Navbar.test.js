import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Navbar from "./Navbar";

describe("Navbar Component", () => {
  it("render correctly", () => {
    const navbarComponent = renderer
      .create(
        <Router>
          <Navbar />
        </Router>
      )
      .toJSON();
    expect(navbarComponent).toMatchSnapshot();
  });
});
