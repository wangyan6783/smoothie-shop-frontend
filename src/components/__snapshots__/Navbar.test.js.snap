// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component render correctly 1`] = `
<nav
  className="navbar"
>
  <ul
    className="navbar-list"
  >
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/smoothies"
        onClick={[Function]}
      >
        Browse
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/"
        onClick={[Function]}
      >
        Our Story
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/"
        onClick={[Function]}
      >
        Gifts
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-logo"
        href="/"
        onClick={[Function]}
      >
        Daily Harvest
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/"
        onClick={[Function]}
      >
        Help
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="navbar-item"
    >
      <a
        className="navbar-link"
        href="/"
        onClick={[Function]}
      >
        Get Started
      </a>
    </li>
  </ul>
</nav>
`;
